libraries/common/subscriptions/router.spec.js

```diff
@@ -208,49 +225,92 @@ describe('Router subscriptions', () => {
         silent: true,
       };
 
-      callback(createCallbackPayload({ params }));
+      await callback(createCallbackPayload({ params }));
 
       testExpectedCall(conductor.push, params);
     });
 
-    it('should redirect to another location correctly', () => {
+    it('should redirect to another location correctly', async () => {
       redirects.set('/some_route', '/some_other_route');
 
       const params = {
         action: ACTION_PUSH,
         pathname: '/some_route',
       };
 
-      callback(createCallbackPayload({ params }));
+      await callback(createCallbackPayload({ params }));
       testExpectedCall(conductor.push);
       expect(conductor.push).toHaveBeenCalledWith('/some_other_route', params.state, params.silent);
     });
 
-    it('should convert shop links as expected', () => {
+    it('should redirect to another location when the redirect handler is a promise', async () => {
+      /**
+       * @return {Promise}
+       */
+      const redirectHandler = () => Promise.resolve('/some_other_route');
+
+      redirects.set('/some_route', redirectHandler);
+
+      const params = {
+        action: ACTION_PUSH,
+        pathname: '/some_route',
+      };
+
+      await callback(createCallbackPayload({ params }));
+      expect(conductor.push).toHaveBeenCalledTimes(1);
+      expect(conductor.push).toHaveBeenCalledWith('/some_other_route', params.state, params.silent);
+      expect(dispatch).toHaveBeenCalledTimes(2);
+      expect(setViewLoading).toHaveBeenCalledWith(protectorRoute);
+      expect(unsetViewLoading).toHaveBeenCalledWith(protectorRoute);
+    });
+
+    it('should abort navigation when a redirect handler rejects', async () => {
+      const error = new Error('W00ps');
+
+      /**
+       * @return {Promise}
+       */
+      const redirectHandler = () => Promise.reject(error);
```

Comment:
No need to store this into a variable. Please put that inline. (reduces the amount of code)

Reply:
i did it because it makes the test more readable from my point of view. a function which returns a rejected promise as the 2nd argument of the `set` method would be hard to understand if someone would have to understand or extend this test.
so i'd be happy to keep it like it is.